pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -794,6 +794,22 @@ final class Event {
           },
         );
 
+  /// Event that is sent from an IDE plugin.
+  ///
+  /// [name] - the name of the event.
+  ///
+  /// [ide] - the reporting [IDE].
+  ///
+  /// [additionalData] - any additional data.
+  Event.idePluginEvent({
+    required String name,
```

Comment:
I'm guessing that this PR is mostly a way to open a conversation and not something you're planning on merging. I say that because this constructor will never be invoked outside of the test you wrote (because none of the plugins are written in Dart).